src/ViewerOptions.tsx

```diff
@@ -0,0 +1,199 @@
+import React from "react";
+import { PluginContext } from "molstar/lib/mol-plugin/context";
+import { Script } from "molstar/lib/mol-script/script";
+import { StructureSelection } from "molstar/lib/mol-model/structure";
+import { StructureSelectionQuery } from "molstar/lib/mol-plugin-state/helpers/structure-selection-query";
+import { setStructureOverpaint } from "molstar/lib/mol-plugin-state/helpers/structure-overpaint";
+import { MolScriptBuilder as MS } from "molstar/lib/mol-script/language/builder";
+import { Structure } from "molstar/lib/mol-model/structure";
+import { Color } from "molstar/lib/mol-util/color";
+
+interface ViewerOptionsProps {
+    plugin: React.MutableRefObject<PluginContext | null>;
+}
+
+const ViewerOptions: React.FunctionComponent<ViewerOptionsProps> = ({
+    plugin,
+}) => {
+    async function createComponent(
+        event: React.SyntheticEvent<HTMLFormElement>
+    ) {
+        event.preventDefault();
+        const form = event.currentTarget;
+        const formElements = form.elements as typeof form.elements & {
+            start: { value: number };
+            end: { value: number };
+            // colourPicker: { value: string };
+            name: { value: string };
+            style: { value: string };
+        };
+        const range = Array.from(
+            { length: formElements.end.value - formElements.start.value },
+            (v, k) => k + Number(formElements.start.value)
+        );
+        const data =
+            plugin.current!.managers.structure.hierarchy.current.structures[0]
+                ?.cell.obj?.data;
+        if (!data) return;
```

Comment:
What does it mean if data is false-y, should the UI do anything?

Reply:
Added alert (for now)